openkfw/TruBudget
# Fix fetching of history items and bug fixes

# What was changed
## Fetching of history items
The project/subproject/workflowitem history uses infinite scrolling.
Before this commit: When the last page is reached, the limit was not adjusted which lead to
overfetching.
Now: The limit is calculated based on the offsetIndex. If it is 0 (i.e.
the last page is reached), the limit is set to the modulus of the number
of events and the limit, which leaves only the events that were not
fetched before.
This also fixes the problem where the history drawer was populated with the wrong items after closing.

## Bugfixes
* Added stringify entries for update and delete projected budgets
* Fixed stringify of permission related history entries. The "formatPermission" method was restored to translate the permissions from the event to a value that can be read from the language files.

Closes #275 
Closes #279 

# How to test
Create more than 50 (current initial fetching limit) events for project/subproject/workflowitem, open the history and scroll to the bottom. The history items should be correctly loaded and no item should appear more than once. The number of items displayed should match the number of history events. (The number of history events can be checked via the `viewHistory`/`viewHistory.v2` endpoint of each resource.)
Each history entry should have a properly formatted text.
